### PR TITLE
Api: Call function.currentVersion to work around CDK issue

### LINKED ISSUE #1001: Call function.currentVersion to work around CDK issue
See this issue:  https://github.com/aws/aws-cdk/issues/13731#issuecomment-814801449

We should automatically call `function.currentVersion` if `function.currentVersionOptions` exists

### Requests
- https://serverless-stack.slack.com/archives/C01HQQVC8TH/p1636130220420300

### PR DESCRIPTION
**Pull request by junaway**
Fixes #1001.

### CODE REVIEW

File: packages/resources/src/Function.ts [hunk 1]
```diff
@@ -373,6 +373,10 @@ export class Function extends lambda.Function {
     } as FunctionHandlerProps);
 
     this._isLiveDevEnabled = isLiveDevEnabled;
+    
+    // Fixes CDK issue with Provisioned Concurrency and Lambda Versions
+    // cf. https://github.com/aws/aws-cdk/issues/13731#issuecomment-814801449 
+    this.currentVersion;
```
Comment (fwang) [inline packages/resources/src/Function.ts:379]: Don't think we should always calling `this.currentVersion`. @thdxr can you confirm on this?

File: packages/resources/test/Function.test.ts [hunk 2]
```diff
@@ -339,6 +339,23 @@ test("timeout-class", async () => {
   );
 });
 
+test("provisioned-concurrency", async () => {
+  const stack = new Stack(new App(), "stack");
+  const foo = new Function(stack, "Function", {
```
Comment (fwang) [inline packages/resources/test/Function.test.ts:344]: foo never used

File: packages/resources/test/Function.test.ts [hunk 3]
```diff
@@ -339,6 +339,23 @@ test("timeout-class", async () => {
...
+  });
+  expectCdk(stack).to(
+    haveResource("AWS::Lambda::Version", {
+      ProvisionedConcurrencyConfig: {
+        ProvisionedConcurrentExecutions: 1,
+      },
+    })
+  );
+});
+
```
Comment (fwang) [inline packages/resources/test/Function.test.ts:358]: Let's also test `AWS::Lambda::Version` not created if concurrency is not configured.

**Review by 53jk1 — APPROVED** [review]

**Comment by thdxr** [comment]
We are transition this repo to https://github.com/sst/v2 and are closing this issue for now. If it is still relevant please reopen it there - sorry for the trouble

**Pull request closed without merging**